proxy/envoy/config.go

```diff
@@ -111,6 +115,21 @@ func buildConfig(listeners Listeners, clusters Clusters, mesh *proxyconfig.Proxy
 		},
 	}
 
+	if mesh.StatsdUdpAddress != "" {
```

Comment:
config gen is supposed to be a pure function since its output is cached. Can you please move this chunk to `proxy/envoy/watcher.go` close to where agent is instantiated? You'd also need to resolve DNS in `proxy/envoy/ingress.go` and `proxy/envoy/egress.go`. 

If you want to reload Envoy on DNS change, you'd need to add a polling routine and trigger. I suggest not do that and just wait till Envoy starts using UDP clusters.